src/hotspot/share/compiler/compilationPolicy.cpp

```diff
@@ -674,9 +675,8 @@ CompileTask* CompilationPolicy::select_task(CompileQueue* compile_queue) {
 
   methodHandle max_method_h(Thread::current(), max_method);
 
-  if (max_task != NULL && max_task->comp_level() == CompLevel_full_profile &&
-      TieredStopAtLevel > CompLevel_full_profile &&
-      max_method != NULL && is_method_profiled(max_method_h)) {
+  if (max_task != NULL && max_task->comp_level() == CompLevel_full_profile && TieredStopAtLevel > CompLevel_full_profile &&
```

Comment:
What is value of `TieredStopAtLevel` when Tiered is off in case you have only C2 and when you have only C1?

Reply:
The default (i.e. CompLevel_full_optimization == 4). It doesn't matter though on this particular line, because this in-queue change optimization is for CompLevel_full_profile compilation requests only, and that level is illegal with both C1-only and C2-only configuration (including when tiered is off). So, this if condition is never going to be true in these modes.